test-js/unit/pentaho/util/robject.Spec.js

```diff
@@ -0,0 +1,126 @@
+/*!
+ * Copyright 2010 - 2016 Pentaho Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+define(function() {
+  "use strict";
+
+  /*global describe:false, it:false, expect:false, beforeEach:false, beforeAll:false, Object:false*/
+
+  describe("squire pentaho.util.object - ", function() {
+    var Spam, protoEggs, parrot;
+
+    beforeEach(function() {
+      Spam = function() {
+        this.bar = "bar";
+      };
+      Spam.prototype = new (function() {
+        this.spam = "spam";
+      })();
+      parrot = new Spam();
+
+      var myProto = function() {
+        this.spam = "eggs";
+      };
+      protoEggs = new myProto();
+    });
+
+    it("should return the input object, if the desired prototype is an object", function(done){
+      var localRequire = require.newContext();
+
+      localRequire(["pentaho/util/object"], function(O) {
+        expect(O.setPrototypeOf(parrot, protoEggs)).toBe(parrot);
+
+        localRequire.dispose();
+        done();
+      });
+    });
+  });
+
+  describe("squire-mocking pentaho.util.object", function() {
+
+    var Spam, protoEggs, parrot;
+    beforeEach(function() {
+
+      Spam = function() {
+        this.bar = "bar";
+      };
+      Spam.prototype = new (function() {
+        this.spam = "spam";
+      })();
+      parrot = new Spam();
+
+      var myProto = function() {
+        this.spam = "eggs";
+      };
+      protoEggs = new myProto();
+    });
+
+
+    [{
+      _label: "setProtoProp",
+      has: {
+        "Object.setPrototypeOf": false,
+        "Object.prototype.__proto__": true
+      }
+    }, {
+      _label: "ES5",
+      has: {
+        "Object.setPrototypeOf": true,
+        "Object.prototype.__proto__": true
+      }
+    }, {
+      _label: "setProtoCopy",
+      has: {
+        "Object.setPrototypeOf": false,
+        "Object.prototype.__proto__": false
+      }
+    }].forEach(function(conf) {
+
+      it("should return the input object, if the desired prototype is an object - " + conf._label, function(done) {
+
+        // Mocks
+        define("pentaho/shim/env", conf);
```

Comment:
Whatever you `define` just before creating a new require context gets captured into that context, so this doesn't affect the global registration. Alternatively, if this seems strange, the user can specify the "mocks" keyArgs. But there's really nothing we can do against this having this effect.
This also allows all of the features of `define`, comparing to the more limited mocks option.

On rethinking this, since last night, I think the `keyArgs` argument should instead be a full requireJS configuration, with an additional, customly handled property, "mocks".